pkg/core/native/management.go

```diff
@@ -182,15 +216,30 @@ func (m *Management) getContract(ic *interop.Context, args []stackitem.Item) sta
 }
 
 // GetContract returns a contract with the given hash from the given DAO.
-func (m *Management) GetContract(d *dao.Simple, hash util.Uint160) (*state.Contract, error) {
-	cache := d.GetROCache(m.ID).(*ManagementCache)
+func GetContract(d *dao.Simple, hash util.Uint160) (*state.Contract, error) {
+	cache := d.GetROCache(ManagementContractID).(*ManagementCache)
 	cs, ok := cache.contracts[hash]
 	if !ok {
 		return nil, storage.ErrKeyNotFound
 	}
 	return cs, nil
 }
 
+// GetContractByID returns a contract with the given ID from the given DAO.
+func GetContractByID(d *dao.Simple, id int32) (*state.Contract, error) {
+	key := make([]byte, 5)
+	key = putHashKey(key, id)
+	si := d.GetStorageItem(ManagementContractID, key)
```

Comment:
Don't we want to cache ID-Hash pairs inside ManagementCache?

Reply:
I don't see a lot of value in doing this. Regular calls request contracts by hashes and our previous id-hash mapping was not cached, so why bother? It can be added afterwards if it's a problem.